lib/snapshot.go

```diff
@@ -472,13 +530,15 @@ func (snap *Snapshot) Run() {
 
 		case SnapshotOperationChecksumAdd:
 			if err := snap.Checksum.AddOrRemoveBytesWithMigrations(operation.checksumKey, operation.checksumValue,
-				snap.Status.CurrentBlockHeight, snap.Migrations.migrationChecksums, true); err != nil {
+				snap.Status.CurrentBlockHeight, snap.Migrations.migrationChecksums,
+				snap.Migrations.migrationChecksumLock, true); err != nil {
```

Comment:
Interesting that we had to add this lock here. I'll take a closer look at it after gym.